letsencrypt/cli.py

```diff
@@ -740,6 +738,52 @@ def install(config, plugins):
     le_client.enhance_config(domains, config)
 
 
+def _set_by_cli(var):
+    """
+    Return True if a particular config variable has been set by the user
+    (CLI or config file) including if the user explicitly set it to the
+    default.  Returns False if the variable was assigned a default value.
+    """
+    detector = _set_by_cli.detector
+    if detector is None:
+        # Setup on first run: `detector` is a weird version of config in which
+        # the default value of every attribute is wrangled to be boolean-false
+        plugins = plugins_disco.PluginsRegistry.find_all()
+        # reconstructed_args == sys.argv[1:], or whatever was passed to main()
+        reconstructed_args = _parser.args + [_parser.verb]
+        default_args = prepare_and_parse_args(plugins, reconstructed_args, detect_defaults=True)
+        detector = _set_by_cli.detector = configuration.NamespaceConfig(default_args, fake=True)
+        # propagate plugin requests: eg --standalone modifies config.authenticator
+        auth, inst = cli_plugin_requests(detector)
+        if auth:
+            detector.namespace.__setattr__("authenticator", auth)
+        if inst:
+            detector.namespace.__setattr__("installer", inst)
+        # more spammy than just debug
+        logger.debug("Default Detector is %r", detector.namespace)
+
+    try:
+        # Is detector.var something that isn't false?
+        change_detected = detector.__getattr__(var)
+    except AttributeError:
+        logger.warning("Missing default analysis for %r", var)
+        return False
+
+    if change_detected:
+        return True
+    # Special case: we actually want account to be set to "" if the server
+    # the account was on has changed
+    elif var == "account" and (detector.server or detector.dry_run or detector.staging):
+        return True
+    # Special case: vars like --no-redirect that get set True -> False
+    # default to None; False means they were set
+    elif var in detector.store_false_vars and change_detected is not None:
```

Comment:
Yes, though I'd prefer to keep the special case totally separate from the `return False` base case for legibility.